pkg/api/api.go

```diff
@@ -34,6 +36,25 @@ func About(c echo.Context) error {
 	return c.JSON(http.StatusOK, data)
 }
 
+func DeleteCanary(c echo.Context) error {
+	id := c.Param("id")
+
+	canary, err := db.FindCanaryByID(id)
+	if err != nil {
+		return errorResonse(c, err, http.StatusInternalServerError)
+	} else if canary == nil {
+		return errorMsgResponse(c, http.StatusNotFound, "canary with id=%s was not found", id)
+	}
+
```

Comment:
Is this required ? We are just checking if the canary with ID exists or not, but the same can happen in the db.DeleteCanaryByID where it should return a `record not found` if we pass the id directly

Reply:
I did this so the HTTP error is well formed _(with the correct status and message)_.

Or else I'd have to either check the gorm `ErrRecordNotFound` error in pkg/api or just transfer the error message from `db.DeleteCanaryByID` with incorrect HTTP status.